content/essentials/intro-to-reactivity.mdx

```diff
@@ -0,0 +1,171 @@
+Instead of recalculating the output each time a function is invoked, the cached result is returned when the same inputs are encountered again. 
+
+By remembering the last computed value and the conditions (or signal values) under which it was computed, the system can avoid unnecessary recalculations.
+
+#### Dependency tracking
+
+An essential aspect of effective memoization in a reactive system is accurate dependency tracking.  
+This involves keeping track of the signals upon which a computed value relies. 
+This way, the system can identify when the cached value needs to be invalidated and recomputed. 
+
+This fine-grained dependency tracking ensures that the system's reactivity remains reliable and accurate.
+
+### Store
+
+Stores are important as they provide a structured and centralized approach to state management. 
+
+A store in reactive programming is essentially a container for state. 
+Unlike dealing with individual signals, stores offer a more organized way to manage state, especially in complex applications.
+
+#### State management
+
+At their core, stores serve as containers for state. 
+They facilitate centralized management of state by offering methods to update, reset, or manipulate it. 
+This not only improves the predictability of state transitions but simplifies the process of interacting with and modifying state.
+
+When the state within a store changes, it triggers a cascade of reactive updates throughout the system. 
+Components or functions that are subscribed to a store are promptly notified of these changes, where they are able to then respond accordingly. 
+This keeps the user interface and other system components synchronized with the underlying data.
+
+#### Scalability
+
+As applications grow in complexity, managing individual signals can become challenging. 
+Stores provide a scalable solution by allowing related state to be grouped together. 
+
+## Tracking
+
+At its core, tracking is about registering subscriptions. 
+When a signal changes, all subscribed effects are re-run. 
+This enables the automatic propagation of changes throughout an application's components.
+
+Some key points to understand about tracking:
+
+- Tracking subscriptions occur when a signal, or store value, is accessed within an effect. 
+In other words, when you read the value of a signal inside an effect, that effect becomes subscribed to that signal.
+
+- Whenever an effect is created and accesses a signal, it establishes a subscription to that signal. 
+This means that the effect will be re-executed whenever the signal changes.
+
+- Tracking is scoped to effects. 
+If you access a signal outside the context of an effect, no tracking or subscription will occur. 
+This is because there is no effect subscribed to that signal.
+
+- The tracking concept extends to reactive props and store values, as well. 
+When you access a reactive value within an effect, it triggers tracking and subscription just like within signals.
```

Comment:
This is maybe worth mentioning as is, but it's important to understand that only the "signal" exists that can be depended upon, "store values", "reactive pros" etc. don't really exist on their own, like it 100% boils down to whether a signal is called inside an effect/memo or not. How that signal is called is entirely irrelevant.